pipelines/nextflow/modules/utils/utils.nf

```diff
@@ -50,3 +50,38 @@ def parse_list_param(String multi_value = '', List<String> allowed_values, Boole
     }
     return all_params
 }
+
+def generate_url(protocol, host, port, user, pass=null, database=null) {
+    // Generate a URL when all the credentials are provided
+    base_url = "${protocol}://${user}@${host}:${port}"
+
+    if (pass) {
+        base_url = "${protocol}://${user}:${pass}@${host}:${port}"
+    }
+    if (database) {
+        base_url += "/${database}"
+    }
+    return base_url
+}
+
+
+def extractMySQLArguments(urlString) {
+    //Extract the MySQL arguments from the url
+    def remove_protocol = urlString.split("//")[1]
+
+    def urlparts = remove_protocol.split("@")
+    def user_info = urlparts[0].split(":")
+    def user = user_info[0]
+    def pass = user_info.length > 1 ? user_info[1]:null
+
+    def host_database = urlparts[1].split("/")
+    def host_info = host_database[0].split(":")
+    def host = host_info[0]
+    def port = host_info[1]
+
+    def database = host_database.length > 1 ? host_database[1] : null
```

Comment:
```suggestion
    def url_parts = url.tokenize('/@')
    def user_info = url_parts[1].split(":")
    def user = user_info[0]
    def pass = user_info.length > 1 ? user_info[1] : null
    def (host, port) = url_parts[2].split(":")
    def database = url_parts.length > 3 ? url_parts[3] : null
```
This is just an alternative code to avoid too many splits and index manipulation, but happy to leave yours (although in that case I would suggest to change `urlparts` to `url_parts` and add spaces in `user_info[1] : null`).